gear-programs/vft-manager/app/src/services/mod.rs

```diff
@@ -0,0 +1,479 @@
+            gear_bridge_builtin,
+            eth_client,
+            config,
+        }
+    }
+}
+
+#[derive(Debug, Decode, Encode, TypeInfo, Clone)]
+pub struct Config {
+    gas_to_burn_tokens: u64,
+    gas_for_reply_deposit: u64,
+    gas_to_mint_tokens: u64,
+    gas_to_transfer_tokens: u64,
+    gas_to_process_mint_request: u64,
+    gas_to_send_request_to_builtin: u64,
+    reply_timeout: u32,
+    gas_for_transfer_to_eth_msg: u64,
+    gas_for_event_sending: u64,
+}
+
+impl Config {
+    #[allow(clippy::too_many_arguments)]
+    pub fn new(
+        gas_to_burn_tokens: u64,
+        gas_for_reply_deposit: u64,
+        gas_to_mint_tokens: u64,
+        gas_to_transfer_tokens: u64,
+        gas_to_process_mint_request: u64,
+        gas_to_send_request_to_builtin: u64,
+        reply_timeout: u32,
+        gas_for_transfer_to_eth_msg: u64,
+        gas_for_event_sending: u64,
+    ) -> Self {
+        Self {
+            gas_to_burn_tokens,
+            gas_for_reply_deposit,
+            gas_to_mint_tokens,
+            gas_to_transfer_tokens,
+            gas_to_process_mint_request,
+            gas_to_send_request_to_builtin,
+            reply_timeout,
+            gas_for_transfer_to_eth_msg,
+            gas_for_event_sending,
+        }
+    }
+}
+
+#[service(events = Event)]
+impl<T> VftManager<T>
+where
+    T: ExecContext,
+{
+    pub fn update_erc20_manager_address(&mut self, new_erc20_manager_address: H160) {
+        self.ensure_admin();
+
+        self.state_mut().erc20_manager_address = new_erc20_manager_address;
+    }
+
+    pub fn update_eth_client(&mut self, eth_client_new: ActorId) {
+        self.ensure_admin();
+
+        self.state_mut().eth_client = eth_client_new;
+    }
+
+    pub fn map_vara_to_eth_address(
+        &mut self,
+        vara_token_id: ActorId,
+        eth_token_id: H160,
+        supply_type: TokenSupply,
+    ) {
+        self.ensure_admin();
+
+        self.state_mut()
+            .token_map
+            .insert(vara_token_id, eth_token_id, supply_type);
+
+        self.notify_on(Event::TokenMappingAdded {
+            vara_token_id,
+            eth_token_id,
+        })
+        .expect("Failed to emit event");
+    }
+
+    pub fn remove_vara_to_eth_address(&mut self, vara_token_id: ActorId) {
+        self.ensure_admin();
+
+        let eth_token_id = self.state_mut().token_map.remove(vara_token_id);
+
+        self.notify_on(Event::TokenMappingRemoved {
+            vara_token_id,
+            eth_token_id,
+        })
+        .expect("Failed to emit event");
+    }
+
+    pub fn update_config(&mut self, config: Config) {
+        self.ensure_admin();
+
+        unsafe {
+            CONFIG = Some(config);
+        }
+    }
+
+    fn ensure_admin(&self) {
+        if self.state().admin != self.exec_context.actor_id() {
+            panic!("Not admin")
+        }
+    }
+
+    pub async fn submit_receipt(&mut self, receipt_rlp: Vec<u8>) -> Result<(), Error> {
```

Comment:
Perhaps add a comment for this method. After renaming, its functionality is not immediately obvious.